### ISSUE TITLE
Doom package manager prompts user countless times on sync/upgrade due to .DS_Store files, I have a temporary solution…

### ISSUE THREAD

**Issue by drawingthesun** [doomemacs/doomemacs#7447]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

## The Issue

When running sync or upgrade I often get a prompt regarding a dirty working tree for particular packages, the culprit being the absolute worst file in the history of all files: The dreaded `.DS_Store` Finder insists on creating for every folder the user has access to.

The message is this:

```
doom sync
or
doom sync -u
…
…
> Updating recipe repos...
    > Repository "<package_repo>" has a dirty worktree:

      ?? .DS_Store

         1) Abort
         2) Stash changes
         3) Discard changes (Choose this if unsure)

      How to proceed? (1, 2, 3; don't know? Pick 3) 3
      Discard all local changes permanently? (y or n) 
```

With `doom sync` the issue is somewhat annoying, as I will have to answer these prompts a handful of times depending on my changes.

WIth `doom sync -u` and `doom sync -u -p` the issue is considerably worse, recently after many changes I was looking at over 400 prompts due to `.DS_Store` being created in the package repo folders (inside the `straight` folder in `.local`

## Attempts to fix the issue

I tried using the flag `--force` but this only skipped the *'How to proceed?'* prompt but still requiring user input for the *'Discard all local changes permanently?'* so that cuts the workload in half, but that still is not viable, I would have to spend hours running `doom sync -u -p --force

## My current solution that seems to work

I open another terminal window and run this bash one liner whilst I run the `doom sync …` command.

```bash
while true; do find /Users/<username>/.emacs.d -type f -iname '.DS_Store' -delete -print; sleep 0.01; done
```

This loop catches the `.DS_Store` files quickly and it seems macOS/Finder gives up for a time recreating the files. This allowed me to complete my doom sync with no issues at all.

Even if we cannot get a fix as I imagine this isn't a high priority issue, maybe add bash one liner (or come up with a better one) to the documentation ao that other users with this issue have a somewhat crude temporary fix.

### Current behavior

When I run `doom sync -u -p` the package manager is prompting the user for action on every occurrence of `.DS_Store` files generated by macOS/Finder.

An ideal solution might be to:

1. Check to see if the package manager is running on macOS.
2. If so, look into the repo folders when performing update tasks/sync tasks and check if there is a `.DS_Store` file
3. If the `.DS_Store` exists, check the repo's git status to see if it matches a committed `.DS_Store` (some repo's have inadvertingly already added `.DS_Store`)
4. If the file is identical then do nothing, there should be no issue.
5. If the files exists in the repo's git but the users macOS has made updates to the local version, then revert the local version to the one in the git repo.
6. If the `.DS_Store` does not exist in the packages git repo and exists in the local folder, then delete it immediately prior to any further operations on that package's git repo. (So complete these `.DS_Store` cleanup checks before update commands)

As long as the `.DS_Store` is fixed to match the packages git repo state and then immediately run other commands on the repo, that should suffice as the `.DS_Store` are not auto generated too often when the user is not in the folder and it's a script doing the work.

### Steps to reproduce

1. Open terminal
2. Run either: `doom sync`, `doom sync -u` or `doom sync -u -p` or another doom command that performs similar function.
3. If you are running macOS you may notice this issue like I have.

### System Information

https://pastebin.com/Q0fSj0xF

**Comment by maikol-solis** [comment]
I have the same behavior. My solution is just to do this 

```
find ~/.emacs.d -name .DS_Store -print0 | xargs -0 rm -v &&  sh ~/.emacs.d/bin/doom sync -u
```

and change it to any doom command (purge, clean, doctor, etc).

**Comment by maikol-solis** [comment]
I had my `.DS_Store` ignore rule in `~/.gitignore`, but passing it to `~/.config/git/ignore` resolved the issue. Thanks. 

I wonder why this rule is not declared globally into https://github.com/doomemacs/doomemacs/blob/master/.gitignore.

**Comment by maikol-solis** [comment]
Thanks. Your solution with `~/.config/git/ignore` works as expected. If this could be included into the doom's core, it will be great.

**Comment by hlissner** [comment]
This was indirectly fixed in 8cafbe4408e7d6c68947a2066e5837379d0983c1, thanks to the new `doom-straight--suppress-confirm-a` advice. `-!`/`--force` should now suppress the `Discard all local changes permanently?`.

Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!
